python/dgl/_dataloading/pytorch/dataloader.py

```diff
@@ -599,9 +600,23 @@ def __init__(self, g, nids, graph_sampler, device=None, use_ddp=False, ddp_seed=
             else:
                 dataloader_kwargs[k] = v
 
-        if device is None:
-            # default to the same device the graph is on
-            device = th.device(g.device)
+        # default to the same device the graph is on
+        device = th.device(g.device if device is None else device)
+        num_workers = dataloader_kwargs.get('num_workers', 0)
+
+        if F.device_type(g.device) == 'cuda' or g.is_pinned():
```

Comment:
You don't need F.device_type here due to dataloaders are only supported in PyTorch. You can use `g.device.type == 'cuda'` directly.